framework/src/engine/consensus/certificate_generation/commit_pool.ts

```diff
@@ -330,26 +342,39 @@ export class CommitPool {
 				...this._nonGossipedCommitsLocal.getByHeight(nextHeight),
 				...this._gossipedCommits.getByHeight(nextHeight),
 			];
-			const nextValidators = singleCommits.map(commit => commit.validatorAddress);
 			let aggregateBFTWeight = BigInt(0);
 
 			// Assume BFT parameters exist for next height
 			const { validators: bftParamValidators, certificateThreshold } =
 				await this._bftMethod.getBFTParameters(methodContext, nextHeight);
 
+			const filteredSingleCommits = singleCommits.filter(commit => {
+				const foundValidator = bftParamValidators.find(v =>
+					v.address.equals(commit.validatorAddress),
+				);
+				if (foundValidator && foundValidator.bftWeight === BigInt(0)) {
```

Comment:
If we perform the edit of line 349, then this would never be true, so it gets redundant.